### PR TITLE
api/v1/search Endpoint 

### PR DESCRIPTION
**Pull request by eneskosr**
Hello,

This PR contains the implementation of search mechanism as stated [here](https://github.com/bounswe/bounswe2018group7/wiki/API-Endpoints#search).

Related issue: #146.

### CODE REVIEW

File: app/backend/hiStoryBackend/api/views/v1/search_views.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+        for post in MemoryPost.objects.all():
+            if text in post.title:
+                if post.id not in mem_list:
+                    mem_list.append(post.id)
+            for story in post.story:
+                if text in story['payload']:
+                    if post.id not in mem_list:
+                        mem_list.append(post.id)
+            if text in post.time['data']:
```
Comment (cburakaygun) [inline app/backend/hiStoryBackend/api/views/v1/search_views.py]: Time of a Memory-Post can be empty ('{}`). In this case, `time['data'] == None` and it fails.

**Comment by eneskosr** [comment]
@cburakaygun Of course, these ids not for end users. I thought returning full memory post is not necessary. In frontend, they can retrieve related memory posts with these ids.

File: app/backend/hiStoryBackend/api/views/v1/search_views.py (hunk 1 above)
Comment (eneskosr) [reply app/backend/hiStoryBackend/api/views/v1/search_views.py]: thanks for the warning, i have added control before accessing.

**Comment by cburakaygun** [comment]
I don't think sending a request for every Memory-Post contained in the search result is a good idea. It would be better for frontend to show the users titles and only sending a request for the Memory-Post when the user click on a title.

**Comment by eneskosr** [comment]
Okay, it does make sense. I have changed the endpoint.

**Comment by eneskosr** [comment]
I am merging it, thanks for your reviews.

**Pull request merged**